server/schedule/placement/rule_manager.go

```diff
@@ -263,7 +263,13 @@ func (m *RuleManager) GetRulesForApplyRegion(region *core.RegionInfo) []*Rule {
 // FitRegion fits a region to the rules it matches.
 func (m *RuleManager) FitRegion(stores StoreSet, region *core.RegionInfo) *RegionFit {
 	rules := m.GetRulesForApplyRegion(region)
-	return FitRegion(stores, region, rules)
+	filteredRules := make([]*Rule, 0, len(rules))
+	for _, rule := range rules {
+		if filterRule(rule, stores.GetStores()) {
+			filteredRules = append(filteredRules, rule)
+		}
+	}
```

Comment:
I think it is a good idea but may be the wrong place to do it. Because the returned `RegionFit` will not include the rules dropped here. I'd like to sugguest to move the logic to some place inside the `fitWorker`.

Reply:
I think it's ok that `RegionFit` will not include the dropped rules. What's the difference?

Reply:
updated.